### ISSUE TITLE
AntiFire Hack

### ISSUE THREAD

**Issue by Platinteufel** [NoCheatPlus/Issues#82]
There's a new hack in "Wurst client" called "AntiFire", where ppl doesn't get any damage from fire and lava. I don't have any information about this hack yet, but it works and players can use it with NoCheatPlus 3.13.6-BETA-sMD5NET-b924 installed.

[03:07:26 INFO]: ---- Version information ----
[03:07:26 INFO]: #### Server ####
[03:07:26 INFO]: git-Spigot-db6de12-07c3001 ~MC: 1.8.8~
[03:07:26 INFO]:   detected: 1.8.8
[03:07:26 INFO]: #### NoCheatPlus ####
[03:07:26 INFO]: Plugin: 3.13.6-BETA-sMD5NET-b924
[03:07:26 INFO]: MCAccess: 1.8.4-1.8.8 / Spigot-CB-1.8_R3
[03:07:26 INFO]: Features:
[03:07:26 INFO]:   blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8
[03:07:26 INFO]:   checks: FastConsume | Gutenberg | AttackFrequency | FlyingFrequency | KeepAliveFrequency
[03:07:26 INFO]:   defaults: pvpKnockBackVelocity
[03:07:26 INFO]:   packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance
[03:07:26 INFO]: Hooks: AllViolations~NCP~ 1.0
[03:07:26 INFO]: #### Related Plugins ####
[03:07:26 INFO]: ProtocolLib v3.6.5-SNAPSHOT-b218

**Comment by MyPictures** [comment]
Thanks for the report, going to try out soon.

**Comment by asofold** [comment]
Apart from debug logs WITH ProtocolLib :) , i would like to clarify, which is it:
* Does this mean they touch lava/fire and don't receive damage?
* Does this mean they avoid lava/fire to not move inside in the first place?

**Comment by jettrainz** [comment]
@asofold I have this issue too. When in fire/lava they take the normal amout of damage. To bypass NCP, the player MUST stand still in a safe place (eg not lava or fire) while the player is on fire to not take damage.

**Comment by iProdigy** [comment]
This is an old cheat that is already detected by NoCheatPlus. AntiFire is accomplished by sending an excess of 0x03 packets which results in increased player ticking. As a result, the fire ticks remaining on the player decrease near instantly, giving the effect of AntiFire. When NCP is combined with ProtocolLib, it [monitors the rates of packets like 0x03, 0x04, 0x05, and 0x05](https://github.com/NoCheatPlus/NoCheatPlus/blob/master/NCPCompatProtocolLib/src/main/java/fr/neatmonster/nocheatplus/checks/net/protocollib/MovingFlying.java#L56-L61), effectively blocking this cheat.

**Comment by asofold** [comment]
They could use a packet burst when standing still. Further optimize with faking lag and then burst. FlyingFrequency could trigger, if they do this while moving.

With 1.9 these ticking things might actually be gone.

**Comment by iProdigy** [comment]
Good point. I think it may be worth refining the More Packets check to be able to detect artificial packet bursts (would also prevent traditional implementations of Blink). However, this could be difficult to implement correctly without unjustly penalizing players with high latency.

**Comment by asofold** [comment]
Isn't fast ticking history since 1.9.x? Could someone confirm or deny this, please :). 

If 1.9.x fixes that, it'll be unlikely that we'll add an extra check for past versions. Since we still in theory support past versions down to 1.4.5-R1.0, something could be done, if anybody has a good idea.

Currently the morepackets(player) implementations keeps track of a configurable amount of seconds, using buckets of idk 1/3rd of a second resolution. It will attempt to smoothen some bursts by brushing

**Comment by Platinteufel** [comment]
Now you got instant kicked for "sending too many packets" if you walk into fire with this hack.
